server/src/main/java/org/opensearch/action/bulk/TransportBulkAction.java

```diff
@@ -538,6 +539,11 @@ protected void doRun() {
                 if (docWriteRequest == null) {
                     continue;
                 }
+
+                if (addFailureIfIndexAppendOnlyAndOpsDeleteOrUpdate(docWriteRequest, i, concreteIndices, metadata)) {
```

Comment:
Should this be checked for only after confirming index is not available through `addFailureIfIndexIsUnavailable`?
This will help short circuit for index unavailable cases faster.